novaboot

```diff
@@ -579,13 +585,21 @@ sub generate_grub2_config($$$$;$$)
 	    $first = 0;
 	    my ($kbin, $kcmd) = split(' ', $_, 2);
 	    $kcmd = '' if !defined $kcmd;
-	    print $fg "  multiboot ${base}$kbin $kcmd\n";
+	    my $boot_method = "multiboot";
+	    if (exists $variables->{BOOT_METHOD}) {
+	    $boot_method = $variables->{BOOT_METHOD};
+	    }
+	    print $fg "  $boot_method ${base}$kbin $kcmd\n";
 	} else {
 	    my @args = split;
+	    my $module_load_method = "module";
+	    if (exists $variables->{MODULE_LOAD_METHOD}) {
```

Comment:
I don't think that it makes sense to use `module` with with `linux` BOOT_METHOD and `inited` with `multiboot`. If so, it's better not to define an extra variable but imply the value from BOOT_METHOD. Also, when the method is `linux`, an error should be produced when there is more than two loads lines, because I doubt that Linux supports more than 2 init ramdisks.

Reply:
I introduced it as another variable because grub can handle [more variations](https://www.gnu.org/software/grub/manual/grub/grub.html#OS_002dspecific-notes) of kernel/module loading commands than those two - `knetbsd`/`knetbsd_module_elf` and `linux16`/`initrd16`. But I suppose those are not that commonly used. I would change it as you suggested if no response.

Reply:
I don't want novaboot to allow generating "arbitrary" GRUB2 configuration file but requiring users to use a special (novaboot) syntax for it. If you want arbitrary grub config, you can always write
```sh
copy boot/grub/grub.cfg <<EOF
whatever grub configuration
more configuration
EOF
```
Novaboot also tries to be bootloader independent. It's not always completely possible, but generally I don't want novaboot features to be one-to-one copy of GRUB features.

I think that the even the other commands, you mention, can be specified with BOOT_METHOD. Possible values could be `linux`, `multiboot`, `knetbsd` etc. Based on this single value, different grub config would be generated. For now it's not necessary to implement knetbsd support, but if one needs it later, it could be easily added.

Reply:
`MODULE_LOAD_METHOD` removed. `BOOT_METHOD` mentioned in `--grub2` section.